src/core/data/transform.cc

```diff
@@ -239,6 +248,12 @@ void Project::print(std::ostream& out) const
 
 int32_t Project::target_ndim(int32_t source_ndim) const { return source_ndim + 1; }
 
+void Project::return_promoted_dims(std::vector<int32_t>& dims) const
+{
+  auto finder = std::find(dims.begin(), dims.end(), (dim_));
+  if (finder != dims.end()) { dims.erase(finder); }
```

Comment:
Note that you also need to "shift down" everything in `dims` that is larger than `dim_`.

Reply:
yes, that is right